packages/legend-graph/src/models/protocols/pure/v1/V1_PureGraphManager.ts

```diff
@@ -2788,4 +2792,74 @@ export class V1_PureGraphManager extends AbstractPureGraphManager {
       ],
     );
   }
+
+  // ---------------------------------------- Mapping Analysis ----------------------------------------
+
+  private buildModelCoverageAnalyticsInput = (
+    graph: PureModel,
+    mapping: Mapping,
+    clientVersion: string,
+    modelCoverageAnalyticsInput: V1_ModelCoverageAnalyticsInput,
+  ): V1_ModelCoverageAnalyticsInput => {
+    /**
+     * NOTE: to lessen network load, we might need to think of a way to only include relevant part of the pure model context data here
+     *
+     * Graph data models can be classified based on dependency hieararchy:
+     * 1. Building blocks: models that all other models depend on: e.g. domain models, connections, etc.
+     * 2. Consumers: models that depends on other models: e.g. mapping, service, etc.
+     * 3. Unrelated: models that depends on nothing and vice versa: e.g. text
+     *
+     * It would be great if we can provide a way to walk the mapping to select only relevant part, but the problem is we cannot really walk the lambda
+     * object to identify relevant classes yet. so the more economical way to to base on the classification above and the knowledge about hierarchy between
+     * models (e.g. service can use mapping, runtime, connection, store, etc.) we can roughly prune the graph model data by group. Following is an example
+     * for mapping used for execution, but this can generalized if we introduce hierarchy/ranking for model type
+     *
+     */
+    const graphData = this.getFullGraphModelData(graph);
+    const prunedGraphData = new V1_PureModelContextData();
+    prunedGraphData.elements = graphData.elements.filter(
```

Comment:
So let's consolidate the logic of this and the `buildExecutionInput` and call the shared method `buildExecutionModelContextData`